### DISCUSSION TITLE
Easier branding

### DISCUSSION THREAD

**Discussion by geirawsm** [dani-garcia/vaultwarden#2743] [Ideas]
The web-vault is kind of easy to rebrand, at least if you know your css and how to swap images.
But rebranding the outgoing mails is another story as they are hardcoded in the Docker source (referring to the instructions in [Issue: SMTP Templates](https://github.com/dani-garcia/vaultwarden/issues/328)).

Hoping for a solution where you can divert the "static" or "templates" folder to an folder on the host containing the branded mails and images.

**Comment by BlackDex** [comment]
@geirawsm The same kinda goes for the admin and email templates.
https://github.com/dani-garcia/vaultwarden/blob/main/.env.template#L48-L52

There you can define a custom folder, which can be a host mounted volume, just like the web-vault.
All you need to do is to make a copy of the whole contents of https://github.com/dani-garcia/vaultwarden/tree/main/src/static/templates/email.

Then you can edit everything you want for the whole admin and email templates. Do keep in mind that this will not auto-update if we make changes to the emails and admin interface, and thus could cause issues if not kept up-to-date.

There are easy `header` and `footer` files for the mails which is probably what you want to edit, and you probably do not have to touch any other file there besides those.

Hell, you can even just copy only the `email_footer.hbs`, `email_footer_text.hbs` and `email_header.hbs` files, place them in a `/path/to/templates/email` folder, and point the `TEMPLATES_FOLDER` to `/path/to/templates/`, and it will use those files to replace the original build-in ones, and use the defaults for the others.

**Reply by geirawsm** [reply]
Thanks for replying.

What about the branding images that are hosted on the rocket server, any way of making it possible to redirect the "vw_static"-folder to another mounted volume, as referenced to in [email_header.hbs](https://github.com/dani-garcia/vaultwarden/blob/main/src/static/templates/email/email_header.hbs#L84)?

**Reply by BlackDex** [reply]
I would suggest to create a special folder within the web-vault using a volume and use that. Those are accessable dynamically and should provide what you want.

**Comment by geirawsm** [comment]
Oh wow, that didn't even cross my mind, thanks!
